File: src/main/java/com/vielheit/core/service/UserService.java
package com.vielheit.core.service;

import com.vielheit.core.domain.User;
import com.vielheit.core.exception.ApplicationException;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

@Transactional
public interface UserService extends Service {

    Optional<User> create(User user) throws ApplicationException;

    Optional<User> getByEmailAddress(String email) throws ApplicationException;

    Optional<User> getById(Long id) throws ApplicationException;

    Optional<User> updateUser(Long id, User user) throws ApplicationException;
}

File: src/main/kotlin/org/vielheit/core/service/UserService.kt
package org.vielheit.core.service

import org.springframework.stereotype.Component
import org.vielheit.core.domain.User
import org.vielheit.core.repository.UserNodeRepository
import org.vielheit.core.repository.UserRepository
import javax.transaction.Transactional

@Component
@Transactional
class UserService(val userRepository: UserRepository, val userNodeRepository: UserNodeRepository) {
    fun createUser(user: User) {


        userRepository.save(user)
    }
}